src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php

```diff
@@ -124,6 +127,82 @@ protected function json($data, $status = 200, $headers = array(), $context = arr
     }
 
     /**
+     * Returns a BinaryFileResponse object with original or customized file name and disposition header.
+     *
+     * @param File|string $file        File object, path to file or string with content to be sent as response
+     * @param string|null $fileName    File name to be sent to response or null (will use original file name)
+     * @param string      $disposition Disposition of response ("attachment" is default, other type is "inline")
+     *
+     * @return BinaryFileResponse
+     *
+     * @throws \LogicException|\InvalidArgumentException|\RuntimeException
+     */
+    protected function file($file, $fileName = null, $disposition = ResponseHeaderBag::DISPOSITION_ATTACHMENT)
+    {
+        $deleteFileAfterSend = false;
+
+        if (!is_string($file) && !$file instanceof File) {
+            throw new \InvalidArgumentException(sprint('The "%s" method expects first argument to be a string or an instance of "%s"', __METHOD__, File::class));
+        }
+
+        if (is_string($file) && file_exists($file)) {
+            $file = new File($file);
+        }
+
+        if (is_string($file)) {
+            if ('' === $file) {
+                throw new \InvalidArgumentException('File content can\'t be empty.');
+            }
+
+            if (empty($fileName)) {
+                throw new \InvalidArgumentException('File name can\'t be empty.');
+            }
+            $cacheDir = $this->container->get('kernel')->getCacheDir();
+            $tmpName = md5($fileName);
+            $tmpPath = $cacheDir.DIRECTORY_SEPARATOR.$tmpName;
+
+            try {
+                file_put_contents($tmpPath, $file);
+            } catch (\Exception $e) {
+                throw new \RuntimeException(sprintf('Temporary file "%s" couldn\'t be created.', $tmpPath), 0, $e);
+            }
+
+            $deleteFileAfterSend = true;
+            $file = new File($tmpPath);
+        }
+
+        if (!$file->isReadable()) {
+            throw new \RuntimeException(sprintf('"%s" is not readable.', $file->getPath()));
+        }
+
+        $response = new BinaryFileResponse($file);
+        $mimeType = $file->getMimeType();
```

Comment:
The Content-Type is already set by default here : https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpFoundation/BinaryFileResponse.php#L197